textacy/tm/topic_model.py

```diff
@@ -292,3 +300,135 @@ def topic_weights(self, doc_topic_matrix):
+                ``sklearn.feature_extraction.text.CountVectorizer.get_feature_names()``,
+                or a mapping of term id: term string
+            topics (seq(int) or int, optional): topic(s) to include in termite plot;
+                if -1, all topics are included
+            sort_topics_by ({'index', 'weight'}, optional):
+            highlight_topics (seq(int) or int, optional): indices for up to 6 topics
+                to visually highlight in the plot with contrasting colors
+            n_terms (int, optional): number of top terms to include in termite plot
+            rank_terms_by ({'topic_weight', 'corpus_weight'}, optional): value used
+                to rank terms; the top-ranked ``n_terms`` are included in the plot
+            sort_terms_by ({'seriation', 'weight', 'index', 'alphabetical'}, optional):
+                method used to vertically sort the selected top ``n_terms`` terms;
+                the default ("seriation") groups similar terms together, which
+                facilitates cross-topic assessment
+            save (str, optional): give the full /path/to/fname on disk to save figure
+
+        Returns:
+            ``matplotlib.axes.Axes.axis``: axis on which termite plot is plotted
+
+        Raises:
+            ValueError: if more than 6 topics are selected for highlighting, or
+                an invalid value is passed for the sort_topics_by, rank_terms_by,
+                and/or sort_terms_by params
+
+        References:
+            .. Chuang, Jason, Christopher D. Manning, and Jeffrey Heer. "Termite:
+                Visualization techniques for assessing textual topic models."
+                Proceedings of the International Working Conference on Advanced
+                Visual Interfaces. ACM, 2012.
+            .. for sorting by "seriation", see https://arxiv.org/abs/1406.5370
+
+        .. seealso:: :func:`viz.termite_plot <textacy.viz.termite.termite_plot>`
+
+        TODO: `rank_terms_by` other metrics, e.g. topic salience or relevance
+        """
+        if highlight_topics is not None:
+            if isinstance(highlight_topics, int):
+                highlight_topics = (highlight_topics,)
+            elif len(highlight_topics) > 6:
+                raise ValueError('no more than 6 topics may be highlighted at once')
+        else:
+            highlight_topics = tuple()
+
+        # get topics indices
+        if topics == -1:
+            topic_inds = tuple(range(self.n_topics))
+        elif isinstance(topics, int):
+            topic_inds = (topics,)
+        else:
+            topic_inds = tuple(topics)
+
+        # get topic indices in sorted order
+        if sort_topics_by == 'index':
+            topic_inds = sorted(topic_inds)
+        elif sort_topics_by == 'weight':
+            topic_inds = tuple(topic_ind for topic_ind
+                               in np.argsort(self.topic_weights(self.transform(doc_term_matrix)))[::-1]
```

Comment:
`.fit()` is used to train (sklearn) models, `.transform()` is for applying them to data. (there's also `.fit_transform()` for doing both in one go, more efficiently.)